asdf/tests/test_generic_io.py

```diff
@@ -360,6 +360,7 @@ def test_unicode_open(tmpdir, small_tree):
                 pass
 
 
+@pytest.mark.filterwarnings("ignore:unclosed file .*")
```

Comment:
Is it a bug that the following (from line 387 in this test) leaves an unclosed file?
```
with generic_io.get_file(sys.__stdout__, "w"):
    pass
```

Reply:
No, the issue is this part of the test: https://github.com/WilliamJamieson/asdf/blob/8d872b0703792871cdc440e09c051c35ef7846d6/asdf/tests/test_generic_io.py#L368-L371 leaves the file open

Reply:
I'm still a bit puzzled by this one. Any clue why the open to __stdout__ is in the invalid_obj test in the first place? It does appear that get_file has a close argument that if added to line 387 removes the need for the ignore filter. If the intention is to test the unclosed file perhaps that should be pulled out into a separate test.

Reply:
First, the test is not intended to test the unclosed file. It clearly is intended to to test that an error is raised when opening invalid files. As I noted above, the `get_file` does not even attempt to close files and so therefore it would be expected that this test would leave behind unclosed files.

I do agree that the part of the test you reference should be somewhere else as it does succeed without error. However, that is a red hearing with respect to the changes in this PR. Feel free to open a separate PR moving that part of the test into another test or its own.